lib/core.js

```diff
@@ -35,12 +35,12 @@ function Promise(fn) {
       deferreds.push(deferred)
       return
     }
+    var cb = state ? deferred.onFulfilled : deferred.onRejected
+    if (cb === null) {
+      (state ? deferred.resolve : deferred.reject)(value)
+      return
+    }
     asap(function() {
-      var cb = state ? deferred.onFulfilled : deferred.onRejected
-      if (cb === null) {
-        (state ? deferred.resolve : deferred.reject)(value)
-        return
-      }
```

Comment:
I'm not sure that ordering is specified, though — current behaviour is… […direct callbacks, …dependent promise callbacks], this intermixes them in an arguably more correct order